Gradle Plugin fails when importing protobufs with underscore in filename

I have the following protobuf definitions in `src/main/proto`:

`some_parameter.proto`:
```
syntax = "proto3";

package my.some_parameters;

message SomeParameter {
    string id = 1;
}
```
`my_service.proto`:
```
syntax = "proto3";

package my.some_service;

import "some_parameter.proto";

service MyService {
    rpc DoStuff (my.some_parameters.SomeParameter) returns (my.some_parameters.SomeParameter);
}
```

`gradle.build`:
```
buildscript {
    repositories {
        mavenCentral()
    }
    dependencies {
        classpath "com.google.protobuf:protobuf-gradle-plugin:0.8.3"
    }
}

apply plugin: 'java'
apply plugin: 'com.google.protobuf'

repositories {
    mavenCentral()
}

dependencies {
    compile "com.google.protobuf:protobuf-java:3.4.0"
    compile "io.grpc:grpc-all:1.8.0"
    compile 'com.salesforce.servicelibs:reactor-grpc-stub:0.7.1'
}

protobuf {
    protoc {
        // Download from repositories
        artifact = "com.google.protobuf:protoc:3.4.0"
    }
    plugins {
        grpc {
            artifact = "io.grpc:protoc-gen-grpc-java:1.8.0"
        }
        reactor {
            artifact = "com.salesforce.servicelibs:reactor-grpc:0.7.1:jdk8@jar"
        }
    }
    generateProtoTasks {
        all().each { task ->
            task.plugins {
                grpc {}
                reactor {} // including this fails with:
                // /Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:28: error: package my.some_parameters.SomeParameter does not exist
                // public reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> doStuff(reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> reactorRequest) {
                //                                                                                                                                                        ^
                }
        }
    }
}
```

Gradle error message: 
```
gradle clean build
:clean
:extractIncludeProto
:extractProto UP-TO-DATE
:generateProto
:compileJava
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:28: error: package my.some_parameters.SomeParameter does not exist
        public reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> doStuff(reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> reactorRequest) {
                                                                                                                                                               ^
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:28: error: package my.some_parameters.SomeParameter does not exist
        public reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> doStuff(reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> reactorRequest) {
                                                                           ^
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:36: error: package my.some_parameters.SomeParameter does not exist
        public reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> doStuff(reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> request) {
                                                                                                                                                               ^
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:36: error: package my.some_parameters.SomeParameter does not exist
        public reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> doStuff(reactor.core.publisher.Mono<my.some_parameters.SomeParameter.SomeParameter> request) {
                                                                           ^
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:46: error: package my.some_parameters.SomeParameter does not exist
                                            my.some_parameters.SomeParameter.SomeParameter,
                                                                            ^
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:47: error: package my.some_parameters.SomeParameter does not exist
                                            my.some_parameters.SomeParameter.SomeParameter>(
                                                                            ^
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:73: error: package my.some_parameters.SomeParameter does not exist
                    com.salesforce.reactorgrpc.stub.ServerCalls.oneToOne((my.some_parameters.SomeParameter.SomeParameter) request,
                                                                                                          ^
/Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java:74: error: package my.some_parameters.SomeParameter does not exist
                            (io.grpc.stub.StreamObserver<my.some_parameters.SomeParameter.SomeParameter>) responseObserver,
                                                                                         ^
Note: /Users/dsandjaja/development/projects/testing/build/generated/source/proto/main/reactor/my/some_service/ReactorMyServiceGrpc.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
8 errors
:compileJava FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 1.472 secs
```

If I rename `some_parameter.proto` to `someparameter.proto` and change the imports accordingly, compilation and Reactor class generation works.

It also works if I add the following to `some_parameter.proto`:
```
option java_outer_classname = "SomeParameterProto";
```

As it works without the reactor plugin, I would consider this a bug.

--- a/pom.xml
+++ b/pom.xml
@@ -53,7 +53,7 @@
         <!-- Dependency Versions -->
         <grpc.version>1.6.1</grpc.version>
         <protoc.version>3.3.0</protoc.version> <!-- Same version as grpc-proto -->
-        <grpc.contrib.version>0.6.1</grpc.contrib.version>
+        <grpc.contrib.version>0.7.0</grpc.contrib.version>
 
         <!-- Test Dependency Versions -->
         <junit.version>4.12</junit.version>

--- a/rx-java/rxgrpc-test/src/test/proto/some_parameter.proto
+++ b/rx-java/rxgrpc-test/src/test/proto/some_parameter.proto
@@ -1,0 +1,14 @@
+syntax = "proto3";
+
+// Test file for https://github.com/salesforce/reactive-grpc/issues/26
+// If this proto compiles, then the test passes.
+
+package my.someparameters;
+
+message SomeParameter {
+    string id = 1;
+}
+
+service WithParameter {
+    rpc SayGoodbye (SomeParameter) returns (SomeParameter) {}
+}

